Use gnome-power-cmd to suspend the computer instead of the nonexistent 
gdm-signal.

Index: gnome-settings-daemon-2.26.0/plugins/media-keys/gsd-media-keys-manager.c
===================================================================
--- gnome-settings-daemon-2.26.0.orig/plugins/media-keys/gsd-media-keys-manager.c	2009-01-22 13:08:29.000000000 +0100
+++ gnome-settings-daemon-2.26.0/plugins/media-keys/gsd-media-keys-manager.c	2009-04-11 13:38:05.607025529 +0200
@@ -71,7 +71,7 @@
 #if defined(__OpenBSD__) || defined(__FreeBSD__)
 # define SLEEP_COMMAND "zzz"
 #else
-# define SLEEP_COMMAND "apm"
+# define SLEEP_COMMAND "gnome-power-cmd suspend"
 #endif
 
 #define GSD_MEDIA_KEYS_MANAGER_GET_PRIVATE(o) (G_TYPE_INSTANCE_GET_PRIVATE ((o), GSD_TYPE_MEDIA_KEYS_MANAGER, GsdMediaKeysManagerPrivate))
